### DISCUSSION TITLE
Deploy storybook under sub-path (baseUrl)

### DISCUSSION THREAD

**Discussion by Shinigami92** [storybookjs/storybook#25858] [Help]
### Summary

We have created a storybook and deployed behind an nginx. Out of the box everything seems to work, but we have components that needs access to files in the public folder and so from `https://<domain>/storybook/` instead of `https://<domain>/`.

So how can we configure the `baseUrl` to be `/storybook/`?

_We are using Vite, if this is important._
_But the vite.config.ts `base` property did nothing._

**Comment by mehmethalis** [comment]
Hi Shinigami,
Maybe, you can overwrite with the viteFinal method in the .storybook/main.ts storybook config. Related official doc is [here](https://storybook.js.org/docs/api/main-config-vite-final)
Have you tried give ' ' for base path?  or you can customize here
```typescript
...
staticDirs: ['../public'],
async viteFinal(config) {
   config.plugins = await withoutVitePlugins(config.plugins, ['vite:dts'])
    return mergeConfig(config, {
      base: '',
    })
  },

```

**Reply by Shinigami92** [reply]
I tried it and even logged it.
The `mergeConfig` is working and the `base` value changes correctly. But it does not have any effect 🤔 
Nevertheless I'm configuring, it does not start for example under `http://localhost:6006/storybook/`

This `base` property works in other vite projects just fine.

**Reply by Shinigami92** [reply]
I found out that there is an express app.

https://github.com/storybookjs/storybook/blob/next/code/lib/core-server/src/dev-server.ts

I assume this one need to mount the routes under a `baseUrl`.

**Comment by nbkhope** [comment]
I can't get this to work. I've tried the approaches here and in other issues. I've tried asking AI to help me out with this. The changes made don't seem to help / make it work. Does anyone know if it is possible to make storybook static website work on a subpath of a domain such as `https://www.hello.com/storybook-ui/` ?
